coffee/lib/abstract-chosen.coffee

```diff
@@ -201,10 +200,49 @@ class AbstractChosen
       this.update_results_content this.results_option_build()
       this.winnow_results_set_highlight()
 
+  highlight_search_result: (regex, zregex, value, search) ->
+    if @search_fuzzy
+      search_text = []
+
+      tokens = search.split(' ').sort (a, b) ->
+        b.length - a.length
+
+      search_string = value.split ' '
+
+      for string in search_string
+        for token in tokens
+          if string.indexOf('<em>') == -1 && string.indexOf('</em>') == -1
+            string = string.replace(new RegExp('(' + token.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&") + ')', 'gi'), '<em>$1</em>')
+        search_text.push string
+
+      result = search_text.join ' '
+    else
+      startpos = value.search zregex
+      text = value.substr(0, startpos + search.length) + '</em>' + value.substr(startpos + search.length)
+      result = text.substr(0, startpos) + '<em>' + text.substr(startpos)
+
+    result
+
+  get_tokenized_regex_string: (text) ->
+    regex_tokens = []
+    tokens = this.get_tokenized_search_string text
+
+    for text in tokens
+      text = text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&")
+      regex_tokens.push "(?=.*" + text + ")"
+
+    regex_tokens.join ''
+
+  get_tokenized_search_string: (text) ->
+    text.split ' '
+
   get_search_regex: (escaped_search_string) ->
-    regex_anchor = if @search_contains then "" else "^"
-    regex_flag = if @case_sensitive_search then "" else "i"
-    new RegExp(regex_anchor + escaped_search_string, regex_flag)
+    if @search_fuzzy
+      new RegExp(this.get_tokenized_regex_string(escaped_search_string), 'i')
+    else
+      regex_anchor = if @search_contains then "" else "^"
+      regex_flag = if @case_sensitive_search then "" else "i"
+      new RegExp(regex_anchor + escaped_search_string, regex_flag)
```

Comment:
Chosen used to pass an escaped string here. If search_fuzzy is true, the unescaped string is being escaped in `get_tokenized_regex_string`. Else, the unescaped string is being passed directly into the new RegExp – causing an error to be thrown. @StehlikC